src/main/kotlin/org/ionproject/integration/builder/implementations/IselTimetableTeachersBuilder.kt

```diff
@@ -0,0 +1,259 @@
+package org.ionproject.integration.builder.implementations
+
+import com.squareup.moshi.Types
+import java.time.Duration
+import java.time.LocalTime
+import org.ionproject.integration.builder.exceptions.TimetableTeachersBuilderException
+import org.ionproject.integration.builder.interfaces.TimetableTeachersBuilder
+import org.ionproject.integration.model.internal.tabula.Cell
+import org.ionproject.integration.model.internal.tabula.Table
+import org.ionproject.integration.model.internal.timetable.Course
+import org.ionproject.integration.model.internal.timetable.CourseTeacher
+import org.ionproject.integration.model.internal.timetable.Event
+import org.ionproject.integration.model.internal.timetable.Faculty
+import org.ionproject.integration.model.internal.timetable.Label
+import org.ionproject.integration.model.internal.timetable.Programme
+import org.ionproject.integration.model.internal.timetable.School
+import org.ionproject.integration.model.internal.timetable.Teacher
+import org.ionproject.integration.model.internal.timetable.Timetable
+import org.ionproject.integration.model.internal.timetable.TimetableTeachers
+import org.ionproject.integration.model.internal.timetable.Weekdays
+import org.ionproject.integration.model.internal.timetable.isel.RawData
+import org.ionproject.integration.utils.JsonUtils
+import org.ionproject.integration.utils.RegexUtils
+import org.ionproject.integration.utils.Try
+
+class IselTimetableTeachersBuilder() : TimetableTeachersBuilder<RawData> {
+    val SCHOOL_REGEX = "\\A.*"
+    val PROGRAMME_REGEX = "^(Licenciatura|Mestrado).*$"
+    val CLASS_SECTION_REGEX = "\\bTurma\\b: [LM][A-Z+]\\d{2}[DN]"
+    val CALENDAR_TERM_REGEX = "\\bAno Letivo\\b: \\d{4}/\\d{2}-\\b(Verão|Inverno)\\b"
+    val TIME_SLOT_REGEX = "([8-9]|1[0-9]|2[0-3]).(0|3)0"
+    val HEIGHT_ONE_HALF_HOUR_THRESHOLD = 47
+    val HEIGHT_HALF_HOUR_THRESHOLD = 17
+
+    private lateinit var iselTimetableTeachers: Try<TimetableTeachers>
+
+    init {
+        reset()
+    }
+
+    override fun reset() {
```

Comment:
I generally don't like `reset` methods. Why can this instance be single use?

Reply:
Just implemented the `builder pattern`. But I can remove the reset method